airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/AirbyteConnectorRunner.kt

```diff
@@ -51,14 +52,25 @@ class AirbyteDestinationRunner(
 /**
  * Replacement for the Micronaut CLI application runner that configures the CLI components and adds
  * the custom property source used to turn the arguments into configuration properties.
+ *
+ * @param isTest Micronaut's TEST env detection relies on inspecting the stacktrace and checking for
+ * any junit calls. This doesn't work if we launch the connector from a different thread, e.g.
+ * `Dispatchers.IO`. Allow callers to force the test env.
```

Comment:
Maybe add a little more clarity here about why we need to launch an application context in the io dispatcher.